projects/ngx-maplibre-gl/src/lib/map/map.component.ts

```diff
@@ -77,291 +77,243 @@ import { Subscription, firstValueFrom } from 'rxjs';
+  readonly boxZoomEnd = output<MapLibreZoomEvent & EventData>();
+  readonly boxZoomCancel = output<MapLibreZoomEvent & EventData>();
+  readonly webGlContextLost = output<MapContextEvent & EventData>();
+  readonly webGlContextRestored = output<MapContextEvent & EventData>();
+  readonly mapLoad = output<Map>();
+  readonly idle = output<MapLibreEvent & EventData>();
+  readonly render = output<MapLibreEvent & EventData>();
+  readonly mapError = output<ErrorEvent & EventData>();
+  readonly data = output<MapDataEvent & EventData>();
+  readonly styleData = output<MapStyleDataEvent & EventData>();
+  readonly sourceData = output<MapSourceDataEvent & EventData>();
+  readonly dataLoading = output<MapDataEvent & EventData>();
+  readonly styleDataLoading = output<MapStyleDataEvent & EventData>();
+  readonly sourceDataLoading = output<MapSourceDataEvent & EventData>();
+  readonly styleImageMissing = output<
+    {
+      id: string;
+    } & EventData
   >();
-  @Output() mapLoad = new EventEmitter<Map>();
-  @Output() idle = new EventEmitter<MapLibreEvent & EventData>();
-  @Output() render = new EventEmitter<MapLibreEvent & EventData>();
-  @Output() mapError = new EventEmitter<ErrorEvent & EventData>();
-  @Output() data = new EventEmitter<MapDataEvent & EventData>();
-  @Output() styleData = new EventEmitter<MapStyleDataEvent & EventData>();
-  @Output() sourceData = new EventEmitter<MapSourceDataEvent & EventData>();
-  @Output() dataLoading = new EventEmitter<MapDataEvent & EventData>();
-  @Output() styleDataLoading = new EventEmitter<
-    MapStyleDataEvent & EventData
-  >();
-  @Output() sourceDataLoading = new EventEmitter<
-    MapSourceDataEvent & EventData
-  >();
-  @Output() styleImageMissing = new EventEmitter<{ id: string } & EventData>();
 
   get mapInstance(): Map {
     return this.mapService.mapInstance;
   }
 
-  private subscriptions: Subscription[] = [];
-
-  @ViewChild('container', { static: true }) mapContainer: ElementRef;
+  readonly mapContainer = viewChild.required<ElementRef>('container');
 
-  constructor(private mapService: MapService, private elementRef: ElementRef) {
+  constructor() {
     afterNextRender(() => {
-      if (this.preserveDrawingBuffer) {
+      if (this.preserveDrawingBuffer()) {
         // This is to allow better interaction with the map state
         const htmlElement: HTMLElement = this.elementRef.nativeElement;
         htmlElement.setAttribute('data-cy', 'map');
-        this.subscriptions.push(
-          this.mapLoad.subscribe(() => {
-            htmlElement.setAttribute('data-loaded', 'true');
-          })
-        );
-        this.subscriptions.push(
-          this.idle.subscribe(() => {
-            htmlElement.setAttribute('data-idle', 'true');
-          })
-        );
-        this.subscriptions.push(
-          this.render.subscribe(() => {
-            htmlElement.removeAttribute('data-idle');
-          })
-        );
+        this.mapLoad.subscribe(() => {
```

Comment:
Where do we unsubscrive? Maybe add takeuntildestroyed?

Reply:
It's not an observable, so we need to use outputToObservable() to convert into an observable and use takeuntilDestroy. But this is not necessary, Angular destroys it itself: 

![image](https://github.com/maplibre/ngx-maplibre-gl/assets/31312682/2ca5b306-ac56-45fe-b9a5-6c07d76c2df2)

Reply:
Ahh, great, thanks!